app/src/components/base/Button/Button.tsx

```diff
@@ -1,41 +1,44 @@
 import React from 'react';
 import styled from '@emotion/styled';
-import { colors, pxToRem } from '../../../styled-theme';
+import { pxToRem } from '../../../styled-theme';
 
 export type ButtonType = 'submit' | 'reset' | 'button';
 
 export interface ButtonProps {
   readonly children: React.ReactNode;
-  readonly onClick?: () => void;
+  readonly onClick?: React.MouseEventHandler<HTMLButtonElement>;
```

Comment:
Hmm does this work? I was expecting something like

```
  readonly onClick?: (event: React.MouseEventHandler<HTMLButtonElement>) => void;
```

I think these needs do be a function type doesn't it?

Reply:
Yep! It works and doesn't seem to effect any other components.

Reply:
Yeah I guess you're right about that. I'll let @DorianKwan weigh in on this since he probably knows more about this stuff than I do.

Reply:
My gut tells me that we should always want to have our method type be a function for `onClick` handlers in case we don't actually care about the event object

Reply:
If we want to make the type more specific, should we start with this?  
`MouseEventHandler<T = Element> = (event: React.MouseEvent<T, MouseEvent>) => void`

If so, could I pair program with one of you at some point today?  It would be great to know exactly how you both want to move forward as we begin moving components to the ui-kit.

Reply:
@mgibson-casper @Thistle-Strum.  My apologies.

It can be written in two different ways:
```
onClick: (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void
```
or
```
onClick: React.MouseEventHandler<HTMLButtonElement>;
```

I believe the ladder is more accurate